### PR TITLE
Support submodules

### PR DESCRIPTION
**Pull request by mmlb**
I was trying to use this to install https://github.com/ul/kak-tree/ which failed to compile because the submodules were not checked out. I'm sure I could have done git submodule update init --recursive in the do block but this seems like something plug.kak should take care.

Note: I'm using a relatively new arg to git, --recurse-submodules, it was introduced in [v2.13](https://github.com/git/git/blob/master/Documentation/RelNotes/2.13.0.txt#L145). I think its worth using it unless someone brings it up as an issue?

### CODE REVIEW

**Comment by mmlb** [comment]
@SeerLite ^

**Comment by andreyorst** [comment]
I think this is a good addition. When I've developed this, there were no plugins which used submodules, that I knew of. However,
I gave it a bit of though lately, and I wonder if this should actually be done via `do` keyword, because if the plugin updates its submodule to a different hash, I think that local clone will not get this update with a simple `git pull`?

What I essentially mean, that if we use recursive strategy for clone, probably we should use recursive strategy for updating too. What do you think? Submodules is always a pain point for me, I quite often just delete the clone, and do a fresh one, because of submodules get out of sync for some reasons.

I also wonder if we were to introduce this behavior, should it be opt-in? Maybe this should better be done with `recursive` keyword for the `plug` command? I think a very small amount of plugins actually requires this feature, so in case `kak-tree` perhaps whe should be just able to write:

``` sh
plug "ul/kaktree" recursive %{
    # configuration
}
```

File: rc/plug.kak [hunk 1]
```diff
@@ -210,7 +210,7 @@ plug -params 1.. -shell-script-candidates %{ ls -1 ${kak_opt_plug_install_dir} }
                         current_branch=$(git branch | awk '/^\*/ { print $2 }')
                         [ "${current_branch}" != "${checkout}" ] && git fetch >/dev/null 2>&1
                     fi
-                    git checkout ${checkout} >/dev/null 2>&1
+                    git checkout --recurse-submodules ${checkout} >/dev/null 2>&1
```
Comment (mmlb) [inline rc/plug.kak:213]: @andreyorst this will update the local checkout to new hash.

**Comment by mmlb** [comment]
> I also wonder if we were to introduce this behavior, should it be opt-in? Maybe this should better be done with `recursive` keyword for the `plug` command? I think a very small amount of plugins actually requires this feature, so in case `kak-tree` perhaps whe should be just able to write:
> 
> ```shell
> plug "ul/kaktree" recursive %{
>     # configuration
> }
> ```

I'm not very keen on adding configuration, it's nice to not have to go and check if the plugin has submodules to know if the line needs a configuration or not. It seems unnecessary too, git doesn't care if there are no submodules while these flags are present. So this setup always does the right thing.

File: rc/plug.kak (hunk 1 above)
Comment (andreyorst) [reply rc/plug.kak:213]: yes, but this runs during plugin load only when any of `branch`, `tag`, or `commit` keywords are present.

**Comment by andreyorst** [comment]
> I'm not very keen on adding configuration, it's nice to not have to go and check if the plugin has submodules to know if the line needs a configuration or not. It seems unnecessary too, git doesn't care if there are no submodules while these flags are present. So this setup always does the right thing.

Seems reasonable. I'm still concerned about recursive update though.

File: rc/plug.kak (hunk 1 above)
Comment (mmlb) [reply rc/plug.kak:213]: @andreyorst I had missed the `git pull` below. Does that cover all the cases now?

Comment (andreyorst) [reply rc/plug.kak:213]: yes, seems good to me!

**Pull request merged**